modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountV2Test.java

```diff
@@ -768,6 +768,82 @@ public void testCommercePercentageDiscounts() throws Exception {
 			actualPrice5.stripTrailingZeros());
 	}
 
+	@Test
+	public void testCommercePercentageDiscountWithMaximumAmount()
+		throws Exception {
+
+		frutillaRule.scenario(
+			"Apply a percentage price discount with a maximum amount lower " +
+				"than the discount amount"
+		).given(
+			"A product with a base price"
+		).and(
+			"A discount with fixed value"
+		).when(
+			"I try to get the final price of the product"
+		).then(
+			"The final price will be calculated taking into consideration " +
+				"the discount"
+		);
+
+		CommerceCatalog catalog =
```

Comment:
@riccardo-alberti we should name this `commerceCatalog`.
```suggestion
		CommerceCatalog commerceCatalog =
```